CMakeLists.txt

```diff
@@ -87,10 +87,22 @@ find_package(Git)
 message(STATUS "Using pg_config ${PG_CONFIG}")
 
 # Check PostgreSQL version
+if (UNIX)
+execute_process(
+  #COMMAND ${PG_CONFIG} --version
+  # Fix: If some PostgreSQL distribution returns something like PostgreSQL 10.2 (Ubuntu 10.2-1.pgdg16.04+1)
+  # Just strip distribution information out.
+  COMMAND /bin/sh -c "${PG_CONFIG} --version | grep -o -E 'PostgreSQL [0-9]*\.?[0-9]+'"
+  OUTPUT_VARIABLE PG_VERSION_STRING
+  OUTPUT_STRIP_TRAILING_WHITESPACE)
+endif (UNIX)
+
+if (WIN32)
 execute_process(
   COMMAND ${PG_CONFIG} --version
   OUTPUT_VARIABLE PG_VERSION_STRING
   OUTPUT_STRIP_TRAILING_WHITESPACE)
+endif (WIN32)
 
 if (NOT ${PG_VERSION_STRING} MATCHES "^PostgreSQL[ ]+([0-9]+)\\.([0-9]+)(\\.([0-9]+))*$")
```

Comment:
I think the fix should be in this regexp. That would also avoid the need for separate UNIX/Windows handling.